### ISSUE TITLE
Insane battery drain on S20+5G Running Android 11

### ISSUE THREAD

**Issue by magical-flying-corn** [signalapp/Signal-Android#11031]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I recently switched to Signal from WhatsApp, I started noticing insane battery drain which didn't happen when I was using WhatsApp as my primary communications method.

### Steps to reproduce
- Own a S20+ 5G Device (Exynos) with same OS version as the screenshots
- Install Signal
- Leave overnight
- Insane battery drain

**Actual result:** Battery drains insanely fast when not using the app
**Expected result:** Battery shouldn't drain this much..?

### Screenshots

I checked my messages once this morning after leaving the device to charge, I had received a few messages (< 10).
Beside this I have not used Signal at all.
![image](https://user-images.githubusercontent.com/79659041/109193385-f98a5f80-77a0-11eb-9a8d-31e74ea36edd.png)

### Device info

**Device:** Samsung Galaxy S20+ 5G / SM-G986B (Exynos)
**Android version:** 11
**Signal version:** 5.4.7 (had 5.3.12 but all the same)

No root, No hacks, Completely default settings, Play services enabled and all that

![image](https://user-images.githubusercontent.com/79659041/109193093-afa17980-77a0-11eb-96d9-7f9c9607f498.png)
![image](https://user-images.githubusercontent.com/79659041/109193120-b62ff100-77a0-11eb-8750-522a780318fb.png)

### Link to debug log
https://debuglogs.org/c5d4a75103e236b24ec50c6308e286d83e1f006b4faf27b95fceaa364f733e42
(Battery drain is up to 36% at the time of debug log, just from Signal)

**Comment by magical-flying-corn** [comment]
I will be back with a debug log shortly

**Comment by magical-flying-corn** [comment]
Attached debug log.

**Comment by fred777** [comment]
I can somewhat confirm this. I'm a long-time user of Signal for Android and only recently I noticed the same (i.e. almost permanent back ground activity) battery drain issue on my Galaxy Note20 with Android 11. Battery runtime dropped from 2 days down to 1.5 days.

Could this be related to a recent update of Signal or Android?

Whatapp, Threema, and Telegram perform much better w.r.t. battery drain. They only consume 1..3% per day opposed to 12..20% by Signal.

**Comment by fred777** [comment]
It's not permanent. I had a couple of days now with Signal background activity in the order of a few minutes instead of hours.

=> server-side issue (message bursts?)?

**Comment by ffalcon31415** [comment]
I am having the same issue on my LG X Power3 running Android 8.1.0

**Comment by Thomvh** [comment]
Can confirm this issue as well on my OnePlus 6T with Android 10

**Comment by SerinitySW** [comment]
Getting this on my Note 9 with Android 10. Crazy battery consumption. This is a recent thing for me.

https://imgur.com/Fcc5EAa

**Comment by blathers123** [comment]
Happened to me the other day with the latest version. Rebooting my phone fixed the issue.
Android 10, OnePlus 6T, battery optimization for signal app disabled

**Comment by loku0** [comment]
my phone is android 11 in 9min chat it takes 1% battary samsung A51

**Comment by mauvehed** [comment]
Seeing similar on Pixel 4a w/ Android 11. My last full charge was 17 hours ago. I can guarantee you I have not had my screen on and Signal open for 15 of those hours.. 

![image](https://user-images.githubusercontent.com/141381/124410701-a4db9000-dd10-11eb-9b55-315fbd768064.png)

**Comment by greyson-signal** [comment]
Unfortunately the only real way for us to investigate batter issues is by analyzing a system [bug report](https://developer.android.com/studio/debug/bug-report). The bug report contains usage information about *all* apps however, so please do not post it publicly. Instead, if you can, please send a bug report to greyson@signal.org. Thanks!

**Comment by ciconi** [comment]
With the update from 5.20.4 to 5.21.6 my battery problem described in #11535 was solved. I'm back to <10% per day.

**Comment by Shadok** [comment]
Same issue with a OnePlus Nord since Android 11.
@greyson-signal, I sent you a bug report and some screenshots.

**Comment by Arjab** [comment]
I have the same issue on my Xiaomi Redmi Note 4 running an [unofficial LineageOS 18.1](https://forum.xda-developers.com/t/ota-sept-21-2021-4-9-fully-stable-high-perf-rom-android-11-lineageos-18-1-mido-vanilla-by-zeelog.4186179/).
In the last couple of days the Signal app drained my phone's battery over night resulting in a dead phone once and 2 % battery remaining this morning. I'm currently charging my phone, last complete charge is 3 days ago and Android is reporting 89 % battery usage for Signal and the time of usage at 2 days and 8 hours. Here's my discharging curve, which is more than insane: https://imgur.com/vnLeQvf
I just updated Signal to v5.24.7 from the previous version and hope this will fix the issue. Also here's my [debug log](https://debuglogs.org/2ad34267e4d0b3dee911174235646bb35738a61ab78adf9782c054b861b412af), which is now roughly 3 hours after I woke up to an almost dead phone.

**Comment by Thomvh** [comment]
> > Your high battery usage is due to not having play services. When play services aren't present, we have to maintain our own websocket connection. When you have poor network, we repeatedly try to reconnect, which drives up battery usage.
> 
> does that mean it is intended behavior and i have to install play service for low battery usage? Or is the problem being worked on?

Yes, on devices without Google Play Services it is intended behaviour that it uses extra battery because it has to keep open a constant connection with Signal servers. MicroG is able to fix this problem without giving all your data to Google with their privacy friendlier implementation of the push protocol where they strip certain part of data out of the request. But the battery issue at play for most people here is not related to this it is mostly just high CPU. Just like video calling. It literally drained 50% of my battery on a brand new Pixel 6 when video calling with someone. I think/assume this is related to the encryption not being optimal for the CPU or something.

**Comment by Jim-Jaspers** [comment]
I have an Exynos based Samsung Galaxy S10e and since Android 11 signal has been giving me absolutely insane battery drain too!  I'm just running stock Android in this case.

![Screenshot_20211104-003229_Device care](https://user-images.githubusercontent.com/93705729/140284701-5f4532b7-961c-4910-833f-0d300547ebd4.jpg)

**Comment by Rude-Boy** [comment]
> I have an Exynos based Samsung Galaxy S10e and since Android 11 signal has been giving me absolutely insane battery drain too! 

For me the latest version of microG from Magisk has helped.

**Comment by lsgunth** [comment]
I'm seeing a similar thing. For the last few months, on random days, my phone turns into a pocket heater and I lose 20-50% of my battery.  I'm fairly certain at this point it is Signal which is causing the issue but I can't do without the App. When I turn on the CPU usage monitor while heating is in progress I always see an org.thoughtcrime.securesms process at the top. Lately I've found when it's happening the most successful way to switch the heater off is to enter signal and hit back a few times until it closes, then clear the open apps.

Similarities to other issues in this chain: I'm also in a small group chat and it seems to happen more often on days when the group chat is active even if I don't engage with it. I use signal desktop rarely and it does not correlate with the battery drain issues. I have google play services installed and it is functional as far as I know, but I do not log into a google account on my phone. Other apps have no issues with push notifications which I assume use play services.

My phone is rather old: an LG G4 with Android 6.0. The signal app is currently 5.25.8 but I've lived with this issue for a number of upgrades of the app and I'm not really sure when it started (though it did not always happen as I've used Signal without noticing these issues for a long time). 

I do not know how to do a system bug report, but this is a debug log taken shortly after turning my pocket heater off today:

https://debuglogs.org/200bbaa90ee209d7801760af6d828abf120a32d7755aa1c8bb9e574b97efaaed

I'm not familiar with what to look for but I do see a retryable exception and a failure after "surpassing max attempts" right around the time I noticed my phone was heating.

Thanks!

**Comment by tw-hx** [comment]
@lsgunth I may have the same problem as you, high battery usage associated with group chats being visible. I've posted an update in my bug report here:

https://github.com/signalapp/Signal-Android/issues/11398

TL;DR ConversationActivity gets into an infinite pause/resume loop using 100% CPU until you exit a group chat. Some but not all chats affects so likely something in the database triggering it.

**Comment by samirm** [comment]
same issue here on Android 12, Pixel 6. all of my friends in the same group chat report the same issue.

really don't understand why this app in particular kills the battery so fast. could it have something to do with heavy crypto tasks? does it encrypt/decrypt with every message sent? what happens with bursts of messages/reactions/etc?

**Comment by lpanepi** [comment]
I have the same issue on a Samsung S7 after maybe deleting or inactivating some Google services. I tried to re-install Google play but this did not solve the issue that high CPU load caused by Signal is observed and app should be closed. Google play ist active but error message still occurs (and battery drain is observed). Any hint how to fix ?

**Comment by Aravindh09** [comment]
I m facing the same issue on my oneplus 7t pro. Video call is causing phone overheating and also battery drain is around 350%. :(

**Comment by tsipo** [comment]
The situation has deteriorated with version 5.32.15 or 5.32.14. I got the 5.32.15 update on Volla OS (based off Android 10) through the app's own update mechanism. No GSF. 
That phone only runs in the background Signal and Telegram apps. Until the latest update a full charge would last ~8 days, since the update (and Telegram wasn't updated recently) it only lasts ~4 days. I can literally see battery drops about 1% per hour when the phone is at home (wifi) and receives no messages.
I have one linked device - Signal linux desktop app on Ubuntu.
Please review changes in 5.32.15 and 5.32.14 (which was released shortly before). 1% an hour may not sound much to you, but that phone is really not doing much and is on wifi most of the time.

**Comment by MoMMde** [comment]
Having this issue on Pixel 6 ![Screenshot_20220902-192529.png](https://user-images.githubusercontent.com/80107781/188207511-7a9b7d9f-1c04-4da9-b9ba-9d906f9c4c86.png)

**Comment by herrsei123** [comment]
I'm having the same issue on Lineage OS 17.0 / Android 10. No Google services installed. Signal Version is always uptodate.
![Screenshot_20220914-161612_Einstellungen](https://user-images.githubusercontent.com/78570906/190180020-74dbefc9-f0fc-4815-83af-b35595592e45.png).
